### PR TITLE
BASW-75: Expose Creating Recur Line Item Subscription Line As API

### PR DESCRIPTION
**Pull request by vinuvarshith**
** Overview **
This is part of BASW-75. Most of the changes there are done on webform_civicrm. A part of the the requirement is to add a subscription line item. This PR exposes adding a subscription line as an api.

### CODE REVIEW

File: api/v3/ContributionRecurLineItem.php [hunk 1]
```diff
@@ -24,6 +37,19 @@ function civicrm_api3_contribution_recur_line_item_create($params) {
...
 }
 
+/**
+ * ContributionRecurLineItem.createsubscriptionline API
+ *
+ * @param array $params
+ * @return array API result descriptor
+ * @throws API_Exception
+ */
+function civicrm_api3_contribution_recur_line_item_createsubscriptionline($params) {
```
Comment (MiyaNoctem) [inline api/v3/ContributionRecurLineItem.php]: I think it's better to name the action in plural, as this is to create all lines that would apply for a payment plan. eg, `createsubscriptionlines`.

File: api/v3/ContributionRecurLineItem.php [hunk 2]
```diff
@@ -24,6 +37,19 @@ function civicrm_api3_contribution_recur_line_item_create($params) {
...
+  $line = new CRM_MembershipExtras_Hook_PostProcess_RecurringContributionLineItemCreator($params['contribution_recur_id']);
+  $line->create();
+  return civicrm_api3_create_success();
```
Comment (MiyaNoctem) [inline api/v3/ContributionRecurLineItem.php]: Blank line before return.

File: api/v3/ContributionRecurLineItem.php [hunk 3]
```diff
@@ -24,6 +37,19 @@ function civicrm_api3_contribution_recur_line_item_create($params) {
   return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
 }
 
+/**
+ * ContributionRecurLineItem.createsubscriptionline API
+ *
+ * @param array $params
+ * @return array API result descriptor
```
Comment (MiyaNoctem) [inline api/v3/ContributionRecurLineItem.php]: Blank line between @params and @return.

**Comment by omarabuhussein** [comment]
@MiyaNoctem do you know the reason for closing this PR ?

**Pull request closed without merging**